src/main/java/uk/sky/cqlmigrate/Lock.java

```diff
@@ -16,6 +16,7 @@
 
     private final LockingMechanism lockingMechanism;
     private final LockConfig lockConfig;
+    private boolean errorHappened;
```

Comment:
Would prefer it if we could keep this class immutable. Could potentially pass a boolean into the `unlock` method to indicate whether or not an error occurred and then call it at the end of the `try` and in the `catch` block.